src/rustup.rs

```diff
@@ -104,9 +104,29 @@ pub fn available_targets(toolchain: &str, msg_info: &mut MessageInfo) -> Result<
     })
 }
 
+fn version(msg_info: &mut MessageInfo) -> Result<semver::Version> {
+    let out = rustup_command(msg_info, false)
+        .arg("-V")
+        .run_and_get_stdout(msg_info)?;
+
+    if let Some(version) = out
+        .lines()
+        .next()
+        .and_then(|line| line.split_whitespace().nth(1))
+    {
+        semver::Version::parse(version).wrap_err_with(|| "failed to parse rustup version")
+    } else {
+        eyre::bail!("failed to get rustup version")
+    }
+}
+
 pub fn install_toolchain(toolchain: &str, msg_info: &mut MessageInfo) -> Result<()> {
-    rustup_command(msg_info, false)
-        .args(&["toolchain", "add", toolchain, "--profile", "minimal"])
+    let mut command = rustup_command(msg_info, false);
+    command.args(&["toolchain", "add", toolchain, "--profile", "minimal"]);
+    if version(msg_info)? >= semver::Version::new(1, 25, 0) {
+        command.arg("--force-non-host");
+    }
+    command
```

Comment:
I don't think we should do this, but rather always try `--force-non-host`, and if that fails then try without